drf_extra_fields/fields.py

```diff
@@ -93,7 +93,7 @@ class Base64ImageField(Base64FieldMixin, ImageField):
     INVALID_FILE_MESSAGE = _("Please upload a valid image.")
     INVALID_TYPE_MESSAGE = _("The type of the image couldn't be determined.")
 
-    def get_file_extension(self, filename, decoded_file):
+    def get_file_extension(self, filename, decoded_file, header=None):
         extension = imghdr.what(filename, decoded_file)
```

Comment:
When you don't know the file extension. You pass the file extension as a parameter of header such as 'data/png' . 

In case where you are trying to create a generic file uploader and what to save the file as 'uuid + extension'. This helps.